### ISSUE TITLE
Warning CS8524 displayed on custom enum

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/roslyn#61087]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/t/Warning-CS8524-displayed-on-custom-enum/10026759)._

---
[severity:It bothers me. A fix would be nice]
To recreate, in C# (.NET 6):
1. Create a custom enum with 3 options, e.g.:

```
public enum MyEnum {
   value1,
   value2,
   value3
}
```

2. Now, assign it to a variable using pattern matching switch statement:

```
string me = theValue switch {
    value1 => "First",
    value2 => "Second",
    value3 => "Third",
};
```

Now I get the warning.

---
#### Feedback Bot on 4/28/2022, 05:58 PM: 

(private comment, text removed)

---
### Original Solutions
(no solutions)

**Comment by CyrusNajmabadi** [comment]
This warning is correct:

> The switch expression does not handle some values of its input type (it is not exhaustive) involving an unnamed enum value. For example, the pattern '(MyEnum)3' is not covered.	

It tells the user what is wrong, and even gives an example.  If hte user doesn't like those warnings, they can disable them.

**Comment by njsokalski** [comment]
I disagree with the warning being correct. First, it says "unnamed enum value", even though it has been named MeEnum. Also, even though the enum is an int behind the scenes, it would be the converting from 3 to MyEnum that would be an error, if the value really is a MyEnum, then all values are covered. Things may be slightly different for enums with [Flags], but they would not normally be used with switch anyway.

**Comment by CyrusNajmabadi** [comment]
> First, it says "unnamed enum value", even though it has been named MeEnum.

`MyEnum` is hte name of the type, not the value.  The value is unnamed.  That message is correct.

>  it would be the converting from 3 to MyEnum that would be an error, 

There is no error doing this.  This is how enums work in .Net/C# (since 1.0) and is an expected use case for all enums. 

> if the value really is a MyEnum, then all values are covered.

No.  All *named* values are covered.  But all the unnamed values (which are all legal enum values as well) are not covered @njsokalski .
